pkg/visor/hypervisor.go

```diff
@@ -489,6 +490,99 @@ func (hv *Hypervisor) getVisorSummary() http.HandlerFunc {
 	})
 }
 
+type extraSummaryWithDmsgResp struct {
+	extraSummaryResp
+	IsHypervisor bool                           `json:"is_hypervisor"`
+	DmsgStats    *dmsgtracker.DmsgClientSummary `json:"dmsg_stats"`
+}
+
+func (hv *Hypervisor) getVisorsExtraSummary() http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		hv.mu.RLock()
+		wg := new(sync.WaitGroup)
+		wg.Add(len(hv.visors))
+
+		i := 0
+		if hv.visor != nil {
```

Comment:
Is there currently any possiblity of this code executing when `hv.Visor` would be nil?

Reply:
Indeed there isn't. I took the code for getting visors and modified it to have additional info in response. However, we never start hypervisor without visor, except in tests. However, some tests rely on nil response on an "empty" hypervisor. Not sure if it's worth the trouble to change the tests and remove nil-checks everywhere.